src/lib/drivers/driver.js

```diff
@@ -86,6 +90,24 @@ class DriverBase {
     return Promise.reject(new Error('Not implemented'));
   }
 
+  evaluateAsync(asyncExpression) {
+    return new Promise((resolve, reject) => {
+      // Inject the call to capture inspection.
+      const expression = `(function() {var __inspect = inspect;${asyncExpression}})()`;
+
+      this.on('Runtime.inspectRequested', value => resolve(value));
+
+      this.sendCommand('Runtime.evaluate', {
+        expression,
+        includeCommandLineAPI: true,
+        returnByValue: false
+      });
+
+      // If this gets to 15s and it hasn't been resolved, reject the Promise.
+      setTimeout(reject, 15000);
```

Comment:
I mean we cannnnnn, but a Promise that's resolved can't be rejected sooooo.... Okay, fine.